ruby/object_oriented_programming_basics/managing_ruby_projects.md

```diff
@@ -0,0 +1,227 @@
+### Introduction
+
+Perhaps going through the curriculum you were wondering about how programmers structure their projects. Do they really just put their entire code in a single file? If they don't, how are they able to use code across multiple files? Are there agreed-upon rules? Common patterns? What are the Ruby ways of doing things? Maybe you haven't yet thought about that but very shortly knowing how to structure your project is going to help you and others navigate the code you've written.
+
+### Lesson overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+- The benefits of splitting your code into separate files.
+- How to make use of code from a separate file.
+- Namespace and scope of `require`d files.
+- What development dependencies are.
+- What gems are.
+- What Bundler is and why it is useful.
+- What `bundle init` does and what a `Gemfile` is.
+- How to get the most out of `Ruby LSP` VSCode extension.
+
+### Confusion, convention, convenience
+
+Recall how you've worked with the projects in Foundations: the HTML, CSS and JS all lived in separate files, becoming "one" thanks to linking CSS and JS in the HTML file(s). Being able to divide the type of work across multiple files hopefully organized your work better and prevented confusion. Moving forward you can imagine that your files are only going to get longer and more plentiful in number, thus making having some kind of structure necessary. Additionally, you might've already realized how useful git tracking changes per-file together with GitHub is by making it a lot easier to track down a bug if you know that your last commit only affected your JS file.
```

Comment:
Love how you connected this to foundations stuff!

Reply:
Good shows have callbacks for folks that watched the previous seasons. Maybe that works for curriculums too? 😆